Library/Homebrew/test/test_versions.rb

```diff
@@ -54,8 +54,10 @@ def test_comparing_regular_versions
   end
 
   def test_HEAD
-    assert_operator version("HEAD"), :>, version("1.2.3")
-    assert_operator version("1.2.3"), :<, version("HEAD")
+    assert_operator head_version("HEAD"), :>, version("1.2.3")
+    assert_operator head_version("HEAD-abcdef"), :>, version("1.2.3")
```

Comment:
> I suppose that HEAD version equals to any other HEAD version. If we want to get latest HEAD prefix, we can do it on keg level, like I do that in Formula#latest_head_prefix.

I think we should consider making the version comparison take into account `latest_head_prefix` (if it doesn't already) so that we can e.g. `brew upgrade` between HEAD versions. It might also not be a bad idea to mandate a `version` if e.g. `git describe --tags` doesn't return anything so we can compare between HEAD versions and normal versions (but obviously that doesn't need to be part of this PR).